src/components/StatesButton/StatesButton.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import style from './StatesButton.st.css';
+import { Button, ButtonProps, SIZE } from '../Button';
+import { Check } from '../../icons/dist';
+
+const delay = time => new Promise(resolve => setTimeout(resolve, time));
+
+export interface StatesButtonProps extends ButtonProps {
+  disabled: boolean;
+  text: string;
+  dataHook?: string;
+}
+
+interface StatesButtonState {
+  success: boolean;
+}
+
+export class StatesButton extends React.Component<
+  StatesButtonProps,
+  StatesButtonState
+> {
+  public buttonRef = React.createRef<HTMLButtonElement>();
+
+  public state = {
+    success: false,
+  };
+
+  public focus = () => {
+    this.buttonRef.current.focus();
+  };
+
+  public async onActionSuccess() {
+    this.setState({ success: true });
+    await delay(2000);
+    this.setState({ success: false });
+  }
+
+  public render() {
+    const { text, disabled, dataHook, ...rest } = this.props;
+    const { success } = this.state;
+    return (
+      <Button
+        data-hook={dataHook}
```

Comment:
This is not needed. We decided to ditch the camelCase datHook in those new wix-ui lobs